en/mavgen_c/README.md

```diff
@@ -59,16 +60,164 @@ When compiling the project, we recommend that you specify the top-level output d
+
+Returns: `0` if packet decoded. `1` if packet decoded.
+
+
+The code fragment below shows the typical use of this function when reading data from a serial port (`serial`):
+```c
+#include <mavlink.h>
+
+mavlink_status_t status;
+mavlink_message_t msg;
+int chan = MAVLINK_COMM_0;
+
+while(serial.bytesAvailable > 0)
+{
+  uint8_t byte = serial.getNextByte();
+  if (mavlink_parse_char(chan, byte, &msg, &status))
+    {
+    printf("Received message with ID %d, sequence: %d from component %d of system %d", msg.msgid, msg.seq, msg.compid, msg.sysid);
+    // ... DECODE THE MESSAGE PAYLOAD HERE ...
+    }
+}
+```
+
+> **Tip** The [mavlink_helpers.h](https://github.com/mavlink/c_library_v2/blob/master/mavlink_helpers.h) include other parser functions: `mavlink_frame_char()` and `mavlink_frame_char_buffer()`.
+  Generally you will want to use `mavlink_parse_char()` (which calls those functions internally), but reviewing the other methods can give you a better understanding of the parsing process.
+
+
+### Decoding the Payload {#decode_payload}
+
+The message/packet object retrieved in the previous section(`mavlink_message_t`) contains fields in the [MAVLink packet/serialization](../guide/serialization.md) format - including the message id (`msgid`) and the payload (`payload64`).
+
+To get the fields of the *specific message* in the packet you need to further decode the payload.
+This is typically done by providing a `case` statement that maps the *ids* of the messages you wish to decode to appropriate decoder functions.
+The code fragment below shows this for two messages (the first decodes a whole message into a C struct, while the second gets just a single field):
+
+```c
+if (mavlink_parse_char(chan, byte, &msg, &status)) {
+...
+```
+```c
+    switch(msg.msgid) {
+      case MAVLINK_MSG_ID_GLOBAL_POSITION_INT: // ID for GLOBAL_POSITION_INT
+        {
+          // Get all fields in payload (into global_position)
+          mavlink_msg_global_position_int_decode(&msg, &global_position);
+          
+        }
+        break;
+      case MAVLINK_MSG_ID_GPS_STATUS:
+        {
+          // Get just one field from payload
+          visible_sats = mavlink_msg_gps_status_get_satellites_visible(&msg);
+        }
+        break;
+     default:
+        break;
+    }
+```
+```c
+...
+}
+```
+
+The decoder/encoder functions and ids for each message in a dialect can be found in separate header files under the dialect folder.
+The headers are named with a format including the message name (**mavlink\_msg\_*message_name*.h**)
+
+> **Tip** Individual message definitions are pulled in when you call `#include <mavlink.h>`, so you don't need to include these separately.
+
+The main decoding function has the signature format `mavlink_msg_<message_name>_decode()`, and decodes the whole message into a C struct (with fields mapping to the original XML message definition).
```

Comment:
signature incorrect.  elipsis are no good in this case since you've used a keyword :-)

Reply:
Don't really understand this, but as above, it is just a name, not a signature. I've removed signature from the text now so I think the confusion is removed?

Reply:
You've *said* it has this signature - when it clearly doesn't.

Reply:
Yes it did thanks. But following your comment I removed that - i.e. you're seeing the outdated version that I since fixed :-). 

I've further changed this to "The most useful decoding function is named with the pattern".